src/main/java/com/parkhomovsky/bookstore/service/implementation/CartItemServiceImpl.java

```diff
@@ -33,10 +33,11 @@ public class CartItemServiceImpl implements CartItemService {
     public CartItemDto add(Long cartItemId, AddCartItemRequestDto addCartItemRequestDto) {
         Optional<CartItem> possibleCartItem = getCartItemById(cartItemId);
         if (possibleCartItem.isPresent()) {
-            CartItem cartItem = addQuantityToCartItem(possibleCartItem.get(),
+            CartItem cartItem = possibleCartItem.get();
+            CartItem updatedCartItem = addQuantityToCartItem(cartItem,
                     addCartItemRequestDto.getQuantity());
-            cartItemRepository.save(cartItem);
-            return cartItemMapper.toDto(cartItem);
+            cartItemRepository.save(updatedCartItem);
+            return cartItemMapper.toDto(updatedCartItem);
         }
         throw new EntityNotFoundException("No item found with provided cartItemId: "
```

Comment:
why not just put this to `orElseThrow` and remove the `if (possibleCartItem.isPresent())`. It would be more clear as for me

Reply:
not resolved